### ISSUE TITLE
Scrumban

### ISSUE THREAD

**Issue by JakubCerovsky** [CenterSprog/CenterSprog#2]
# Scrumban

Let's imagine a software development team using Scrumban to manage their work. This team combines elements from Scrum and Kanban to stay flexible and efficient. Here's a simplified demonstration of how they work:

### Team: 
The team comprises developers, a product owner, and a Scrum Master who helps with process improvement.

### Board: 
They use a Kanban board, a visual representation of their work, divided into columns:

**Backlog**: This column contains all the tasks that need to be done. The product owner continuously adds and prioritizes items here.

**Ready for Development**: When a task is ready to be worked on, it moves to this column. The team can pull tasks from here when they have capacity.

**In Progress (WIP Limits: 3)**: This column is where active work is happening. They limit themselves to working on a maximum of three tasks at a time to maintain focus and prevent overload.

**Code Review**: After a developer finishes coding, the task moves here for code review by another team member.

**Testing**: Tasks that pass code review go to the testing phase.

**Done**: Tasks that pass testing are considered complete and move to this column.

**Process**:

- **Daily Standup**: The team meets every day for a short standup meeting, just like in Scrum. They discuss what they worked on yesterday, what they're working on today, and any obstacles.

- **Planning**: Instead of fixed-length sprints, they have a continuous planning process. The product owner adds and prioritizes tasks in the backlog. The team pulls tasks from the "Ready for Development" column as they have capacity.

- **WIP Limits**: They strictly follow the WIP limits in the "In Progress" column. If it's full, they can't start new tasks until some are moved to the next stage.

- **Continuous Improvement**: The team regularly reviews their Kanban metrics, such as cycle time and lead time, during retrospectives. They use this data to identify bottlenecks or areas for improvement and adjust their process accordingly.

- **Backlog Refinement**: The product owner and team meet periodically to refine and reprioritize the backlog based on feedback, changing requirements, and new insights.

**Flexibility**: Unlike Scrum, they don't have fixed sprint lengths. They continuously deliver work, and if priorities change, they can adapt quickly. For example, if a critical bug appears, they can address it immediately without waiting for the next sprint.

**Outcome**: This Scrumban approach allows the team to balance structure and flexibility. They have clear priorities (like Scrum) and can react swiftly to changes (like Kanban). The visual board helps everyone understand the status of tasks, and data-driven retrospectives lead to continuous process improvements.

## Backlog

In a Scrumban approach, the backlog is continuously filled and refined. Here's when and how you typically fill in the backlog:

**Initial Backlog**: At the beginning of a project or when transitioning to Scrumban, you start with an initial backlog. This backlog contains high-level ideas, features, or user stories that represent the work to be done. It's often created during project initiation or planning.

**Continuous Refinement**: The product owner and the team continuously refine and update the backlog. This means:

- **Adding New Items**: Whenever new ideas or requirements emerge, they are added to the backlog. These can come from stakeholders, customers, or team members.

- **Prioritization**: Items in the backlog are regularly re-prioritized based on business value, customer feedback, and changing project needs. The product owner ensures that the most important items are at the top.

- **Detailing and Slicing**: As the team gets closer to working on an item, it's broken down into smaller, more detailed tasks or user stories. This happens in backlog refinement sessions.

- **Removing or Deprioritizing**: Items that are no longer relevant or important may be removed or moved lower in the backlog.

**Pulling from the Backlog**: When the team has the capacity to work on new tasks, they pull items from the backlog into the "Ready for Development" column on the Kanban board. This is usually based on the prioritization set by the product owner.

**During Planning**: While Scrumban doesn't have fixed-length sprints like Scrum, some teams may choose to hold regular planning sessions (which can be less formal than Scrum sprint planning). During these sessions, the team reviews the backlog, selects items to work on, and discusses the details of those items.

In essence, the backlog in Scrumban is a dynamic and continuously evolving list of work items. It's not limited to a single planning event; instead, it reflects the current state of work to be done and is adjusted as new information becomes available or priorities change. This flexibility allows teams to adapt quickly to shifting project requirements and customer needs.

## Is a Scrum Master role needed?

The role of a Scrum Master is not strictly needed in a Scrumban approach, as Scrumban combines elements of both Scrum and Kanban and is often more flexible in terms of roles. However, the role of a Scrum Master can still be valuable and relevant in a Scrumban environment, depending on the team's needs and preferences. Here are some considerations:

**Process Facilitation**: In a Scrumban team, a Scrum Master can help facilitate the team's process, just as in Scrum. They can assist with daily standup meetings, retrospectives, and other process-related activities, ensuring that the team follows their chosen approach effectively.

**Continuous Improvement**: The Scrum Master can play a crucial role in promoting a culture of continuous improvement within the team. They can use Kanban metrics and retrospectives to identify bottlenecks or areas for improvement and help the team implement changes.

**Removing Impediments**: While Scrumban aims to be more flexible, there may still be obstacles or impediments that the team encounters. A Scrum Master can help identify and remove these impediments, allowing the team to work more smoothly.

**Coaching and Support**: Scrum Masters often serve as coaches and mentors to team members, helping them understand and embrace Agile principles and practices. This coaching role can be valuable in any Agile-based approach, including Scrumban.

**Team Empowerment**: In a Scrumban context, the Scrum Master may focus more on empowering the team to make decisions and adapt their processes. This aligns with the principles of self-organization in Agile.

**Flexibility**: The Scrum Master's role can adapt to the specific needs and preferences of the team. If the team feels that they are effectively self-managing and handling their processes without the need for a Scrum Master, they may choose to reduce the Scrum Master's involvement.

Ultimately, whether you have a dedicated Scrum Master in a Scrumban team depends on the team's dynamics, the level of Agile maturity, and the team's comfort with self-organization. Some Scrumban teams may opt for a less formal Scrum Master role, while others may find value in retaining a Scrum Master to provide guidance and support. It's important for the team to have open discussions and decide on the best approach for their unique situation
